tools/proofers/processtext.php

```diff
@@ -306,6 +306,82 @@ function leave_spellcheck_mode( $ppage )
     }
 }
 
+function attempt_to_save_as_done($ppage, $text_data, $and_do_another)
+// This is only an attempt, because a daily page limit might block the save,
+// or prevent further saves.
+// If there's a problem, this function does not return to the caller.
+{
+    global $code_url, $pguser, $projectid, $proj_state;
```

Comment:
It would be nice if we passed in file-scope-globals like `$projectid` and `$proj_state`. The fewer global variables we have lying around the better. This isn't always easy, but it appears to be in this case.

Reply:
Eliminated those globals, but via extraction from $ppage rather than passing as separate parameters.

Which makes me wonder: should  `attempt_to_save_as_done` be a method of PPage?

Reply:
That's a really good point. It does feel like PPage is a reasonable place for that.

Reply:
Done.